backend/sublet/serializers.py

```diff
@@ -74,29 +99,44 @@ def parse_amenities(self, raw_amenities):
 
     def create(self, validated_data):
         validated_data["subletter"] = self.context["request"].user
+        images = validated_data.pop("images")
         instance = super().create(validated_data)
         data = self.context["request"].POST
         amenities = self.parse_amenities(data.getlist("amenities"))
         instance.amenities.set(amenities)
         instance.save()
+        # TODO: make this atomic
+        for img in images:
```

Comment:
all validation should be done before saving any data. so here, before creating the images and such, run this code. something like:
```
img_serializers = []
for img in imgs:
    img_serializer = SubletImageSerializer(data={"sublet": instance.id, "image": img})
    img_serializer.is_valid(raise_exception=True)
    img_serializers.append(img_serializer)
[img_serializer.save() for img_serializer in img_serializers]
```